pkg/scaler/machineSetScaler.go

```diff
@@ -108,33 +107,18 @@ func (s *machineSetScaler) EnsureScaleDownNodes(c client.Client, nds drain.NodeD
 		}
 	}
 
-	// MachineSets control workers and infras nodes.
-	originalMachineSets := &machineapi.MachineSetList{}
-	err = c.List(context.TODO(), originalMachineSets, []client.ListOption{
+	//new scaler block to verify upgrade machines scaled down.
+	originalMachines := &machineapi.MachineList{}
+	err = c.List(context.TODO(), originalMachines, []client.ListOption{
 		client.InNamespace(MACHINE_API_NAMESPACE),
-		NotMatchingLabels{LABEL_UPGRADE: "true"},
-	}...)
-	if err != nil {
-		logger.Error(err, "failed to get upgrade extra machinesets")
-		return false, err
-	}
-
-	// Desired replicas should match worker and infra count of nodes.
-	nonMasterNodes := &corev1.NodeList{}
-	err = c.List(context.TODO(), nonMasterNodes, []client.ListOption{
-		NotMatchingLabels{machinery.MasterLabel: ""},
+		client.MatchingLabels{LABEL_UPGRADE: "true"},
 	}...)
 	if err != nil {
-		logger.Error(err, "failed to list nodes")
+		logger.Error(err, "Cannot get a list of extra upgrade machines")
 		return false, err
 	}
 
-	var desiredNodeCount int32 = 0
-	for _, ms := range originalMachineSets.Items {
-		desiredNodeCount += *ms.Spec.Replicas
-	}
-
-	if desiredNodeCount != int32(len(nonMasterNodes.Items)) {
+	if len(originalMachines.Items) != 0 {
```

Comment:
It would be helpful if it logged how many upgrade machines are still waiting to be terminated, if there is more than 0 waiting to be terminated.

Reply:
Added a block to print the machines from machine list using the logger.